lib/sub-tasks/git.js

```diff
@@ -0,0 +1,39 @@
+import isAdded from 'is-git-added';
+import isGit from 'is-git-repository';
+import Listr from 'listr';
+
+import getCommits from '../helpers/getCommits';
+
+const cleanup = () => {
+  const cwd = process.cwd();
+  const commits = getCommits();
+
+  return new Listr([
+    {
+      title: 'check if this directory is a git repository',
+      task: () => {
+        if (!isGit(cwd)) {
+          throw new Error('Error: this is not a git repository... make sure you are in the correct directory');
+        }
+      },
+    },
+    {
+      title: 'check if changes are git added',
+      task: () => {
+        if (commits[0] === '' && !isAdded(cwd)) {
+          throw new Error('Error: no changes... try to git add <files>');
+        }
+      },
+    },
+    {
+      title: 'check if added changes are git commited',
```

Comment:
Wouldn't be that the same as [git-needs-push](https://github.com/JPeer264/node-git-needs-push)? Because it will check if there are some commits. It will also check if you have permission to push to the repo.